src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmTestBrowserCommand.cs

```diff
@@ -106,8 +106,14 @@ protected override async Task<ExitCode> InvokeInternal(ILogger logger)
             // close all tabs before quit is a workaround for broken Selenium - GeckoDriver communication in Firefox
             // https://github.com/dotnet/runtime/issues/101617
             logger.LogInformation($"Closing {driver.WindowHandles.Count} browser tabs before setting the main tab to config page and quitting.");
+            DateTime timeoutThreshold = DateTime.Now.AddSeconds(10);
             while (driver.WindowHandles.Count > 1)
             {
+                if (DateTime.Now > timeoutThreshold)
+                {
+                    logger.LogInformation($"Timeout while trying to close tabs, {driver.WindowHandles.Count} is left open before quitting.");
+                    break;
+                }
```

Comment:
DateTime.Now is using local time so could be affected by time changes, a more robust approach is:

```suggestion
            var cts = new CancellationTokenSource();
            cts.CancelAfter(10000);

            while (driver.WindowHandles.Count > 1)
            {
                if (cts.IsCancellationRequested)
                {
                    logger.LogInformation($"Timeout while trying to close tabs, {driver.WindowHandles.Count} is left open before quitting.");
                    break;
                }
```